File: src/test/java/dev/retrotv/crypt/owe/sha/SHA512Test.java
package dev.retrotv.crypt.owe.sha;

import dev.retrotv.enums.Algorithm;
import dev.retrotv.crypt.owe.OWETest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class SHA512Test extends OWETest {

    @Test
    @DisplayName("SHA512 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(Algorithm.SHA512);
    }

    @Test
    @DisplayName("SHA512 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new SHA512(), Algorithm.SHA512);
    }

    @Test
    @DisplayName("SHA512 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new SHA512());
    }

    @Test
    @DisplayName("SHA512 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new SHA512());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA512Test.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class SHA512Test : OWETest() {
    @Test
    @DisplayName("SHA512 File hash 테스트")
    @Throws(Exception::class)
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.SHA512)
    }

    @Test
    @DisplayName("SHA512 File hash matches 테스트")
    @Throws(Exception::class)
    fun fileHashMatchesTest() {
        fileHashMatchesTest(SHA512(), Algorithm.Hash.SHA512)
    }

    @Test
    @DisplayName("SHA512 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(SHA512())
    }
}
